packages/scanner/src/ruleChecker.ts

```diff
@@ -115,7 +130,27 @@ export default class RuleChecker {
       return;
     }
 
-    const buildFinding = (
+    let appmapConfigDir: string | undefined;
+    {
+      let searchDir: string | undefined = dirname(resolve(appMapFileName));
+      while (!appmapConfigDir) {
+        if (await fileExists(join(searchDir, 'appmap.yml'))) {
+          appmapConfigDir = searchDir;
+        } else {
+          if (dirname(searchDir) === searchDir) break;
+
+          searchDir = dirname(searchDir);
```

Comment:
This will loop forever if it reaches the filesystem root.

Reply:
According to the Node docs `(dirname(searchDir) === searchDir)` will be true at the FS root, which is why I'm testing that above.

Reply:
Ah, right, I missed that! Happens reading the code after a sleepless night, I'm so sorry!